blueflood-elasticsearch/src/main/java/com/rackspacecloud/blueflood/io/ElasticIO.java

```diff
@@ -143,22 +140,25 @@ private IndexRequestBuilder createSingleRequest(Discovery md) throws IOException
     }
     
     public List<SearchResult> search(String tenant, String query) throws Exception {
-        // complain if someone is trying to search specifically on any tenant.
-        if (query.indexOf(tenantId.name()) >= 0) {
-            throw new Exception("Illegal query: " + query);
-        }
-        
+        return search(tenant, Arrays.asList(query));
+    }
+
+    public List<SearchResult> search(String tenant, List<String> queries) throws Exception {
         List<SearchResult> results = new ArrayList<SearchResult>();
-        Timer.Context searchTimerCtx = searchTimer.time();
-        
-        // todo: we'll want to change this once we decide and use a query syntax in the query string.
-        BoolQueryBuilder qb = boolQuery()
-                .must(termQuery(tenantId.toString(), tenant))
-                .must(
-                        query.contains("*") ?
-                                wildcardQuery(metric_name.name(), query) :
-                                termQuery(metric_name.name(), query)
-                );
+        Timer.Context multiSearchCtx = searchTimer.time();
+        queryBatchHistogram.update(queries.size());
+        BoolQueryBuilder qb = boolQuery();
+
+        for (String query : queries) {
+            qb.should(boolQuery()
```

Comment:
@tilogaat `should in boolean query is like an `OR` operation, so no, it does not expect all queries to return results. `must` on the other hand is like an `AND` operation and it expects all the queries inside it to get satisfied.